verification/certifyKey.go

```diff
@@ -483,114 +83,259 @@ func testCertifyKey(d TestDPEInstance, c DPEClient, t *testing.T, simulation boo
 		// Check key returned in command response against certificate
 		checkCertifyKeyResponse(t, leafCert, *certifyKeyResp, hashAlg)
 
-		// Validate that all X.509 fields conform with the format defined in the DPE iRoT profile
-		validateCertifyKeyCert(t, leafCert, params.Flags, params.Label)
-
 		// Ensure full certificate chain has valid signatures
 		// This also checks certificate lifetime, signatures as part of cert chain validation
-		validateLeafCertChain(t, certChain, leafCert)
+		if err = validateLeafCertChain(certChain, leafCert); err != nil {
+			t.Errorf("[ERROR]: %v", err)
+		}
+
+		// Check for basic constraints extension
+		if err = checkBasicConstraints(leafCert, params.Flags); err != nil {
+			t.Errorf("[ERROR]: %v", err)
+		}
+
+		// Check key usage extensions - DigitalSignature, CertSign
+		if err = checkKeyExtensions(leafCert); err != nil {
+			t.Errorf("[ERROR]: %v", err)
+		}
+
+		// Check extended key usage extensions
+		if err = checkExtendedKeyUsages(leafCert); err != nil {
+			t.Errorf("[ERROR]: %v", err)
+		}
+
+		// Check critical UEID and Multi Tcb Info TCG extensions
+		// Check UEID extension
+		if err = checkTcgUeidExtension(leafCert, params.Label); err != nil {
+			t.Errorf("[ERROR]: %v", err)
+		}
+
+		// Check MultiTcbInfo Extension structure
+		if _, err = getMultiTcbInfo(leafCert); err != nil {
+			t.Errorf("[ERROR]: %v", err)
+		}
```

Comment:
Calling single functions and passing a long list of arguments is harder to read, so the functions that were called inside validateCertifyKeyCert() are called directly.